File: Calculator/Calculator/app/src/main/java/com/example/calculator/MainActivity.java
package com.example.calculator;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class MainActivity extends AppCompatActivity {

    private EditText a, b, ketqua;
    private Button btnPlus, btnMinus, btnMulti, btnDivide;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        a = findViewById(R.id.a_variable);
        b = findViewById(R.id.b_variable);
        ketqua = findViewById(R.id.result_variable);

        btnPlus = findViewById(R.id.button_plus);
        btnMinus = findViewById(R.id.button_minus);
        btnMulti = findViewById(R.id.button_multi);
        btnDivide = findViewById(R.id.button_divide);
        IListener l = new IListener();
        btnPlus.setOnClickListener(l);
        btnMinus.setOnClickListener(l);
        btnMulti.setOnClickListener(l);
        btnDivide.setOnClickListener(l);


    }
    class IListener implements View.OnClickListener{
        @Override
        public void onClick(View v){
            int id = v.getId();
            if (a.getText().toString().trim().isEmpty() || b.getText().toString().trim().isEmpty())
            {
                Toast.makeText(MainActivity.this,"Nhập số vào, bướng à ?",Toast.LENGTH_SHORT).show();
                return;
            }
            double numA = Double.parseDouble(a.getText().toString().trim());
            double numB = Double.parseDouble(b.getText().toString().trim());
            try {
                if (id == R.id.button_plus){
                    ketqua.setText(String.format("%.2f",numA + numB));
                }
                if (id == R.id.button_minus){
                    ketqua.setText(String.format("%.2f",numA - numB));
                }
                if (id == R.id.button_multi){
                    ketqua.setText(String.format("%.2f",numA * numB));
                }
                if (id == R.id.button_divide){
                    if (numB == 0)
                    {
                        ketqua.getText();
                        Toast.makeText(MainActivity.this,"Mày thích b = 0 không ?",Toast.LENGTH_SHORT).show();
                        return;
                    }
                    ketqua.setText(String.format("%.2f",numA / numB));
                }
            } catch (Exception e)
            {
                Toast.makeText(MainActivity.this,"Lỗi gì rồi thangngu",Toast.LENGTH_SHORT).show();
            }

        }
    }
}

File: SimpleCalculator/app/src/main/java/com/example/simplecalculator/MainActivity.kt
package com.example.simplecalculator

import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val aInput = findViewById<EditText>(R.id.a_variable)
        val bInput = findViewById<EditText>(R.id.b_variable)
        val resultField = findViewById<EditText>(R.id.result_variable)

        val btnPlus = findViewById<Button>(R.id.button_plus)
        val btnMinus = findViewById<Button>(R.id.button_minus)
        val btnMulti = findViewById<Button>(R.id.button_multi)
        val btnDivide = findViewById<Button>(R.id.button_divide)

        fun getValues(): Pair<Double, Double>? {
            val aText = aInput.text.toString()
            val bText = bInput.text.toString()

            if (aText.isEmpty() || bText.isEmpty()) {
                Toast.makeText(this, "Vui lòng nhập đủ a và b", Toast.LENGTH_SHORT).show()
                return null
            }

            val a = aText.toDoubleOrNull()
            val b = bText.toDoubleOrNull()

            if (a == null || b == null) {
                Toast.makeText(this, "Giá trị nhập không hợp lệ", Toast.LENGTH_SHORT).show()
                return null
            }
            return Pair(a, b)
        }

        // Sự kiện nút cộng
        btnPlus.setOnClickListener {
            val values = getValues() ?: return@setOnClickListener
            val result = values.first + values.second
            resultField.setText(result.toString())
        }

        // Sự kiện nút trừ
        btnMinus.setOnClickListener {
            val values = getValues() ?: return@setOnClickListener
            val result = values.first - values.second
            resultField.setText(result.toString())
        }

        // Sự kiện nút nhân
        btnMulti.setOnClickListener {
            val values = getValues() ?: return@setOnClickListener
            val result = values.first * values.second
            resultField.setText(result.toString())
        }

        // Sự kiện nút chia
        btnDivide.setOnClickListener {
            val values = getValues() ?: return@setOnClickListener
            if (values.second == 0.0) {
                Toast.makeText(this, "Không thể chia cho 0", Toast.LENGTH_SHORT).show()
            } else {
                val result = values.first / values.second
                resultField.setText(result.toString())
            }
        }
    }
}
